src/main/java/org/programmers/springboot/basic/domain/customer/service/validate/EmailValidator.java

```diff
@@ -0,0 +1,19 @@
+package org.programmers.springboot.basic.domain.customer.service.validate;
+
+import org.programmers.springboot.basic.domain.customer.exception.IllegalEmailException;
+import org.springframework.stereotype.Component;
+
+import java.util.regex.Pattern;
+
+@Component
+public class EmailValidator {
+
+    public void validate(String email) {
+
+        String regex = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$";
+        Pattern pattern = Pattern.compile(regex);
+        if (!pattern.matcher(email).matches()) {
+            throw new IllegalEmailException();
+        }
+    }
```

Comment:
```suggestion
    private static final String REGEX = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$";
    private static final Pattern pattern = Pattern.compile(REGEX);
    
    public void validate(String email) {
        if (!pattern.matcher(email).matches()) {
            throw new IllegalEmailException();
        }
    }
```

Bean으로 등록하는 기준이 있으실까요?
Pattern 은 상수로 선언하고, 메서드에서 재활용하는 방식으로 선언하면 보다 효과적으로 사용할 수 있습니다.
이는 이펙티브 자바 아이템 6을 참고해보시면 좋습니다.

Reply:
Pattern은 재사용성이 높은 부분이니 메서드 안에 넣는 것보다 static final을 사용해서 재활용하도록 하는 방식이 더 적합한 것 같습니다.

Bean으로 등록할 것인지, static을 이용한 유틸 클래스로 사용할 것인지에 대한 명확한 기준이 아직 잡히지 않아서 해당 부분을 공부하고 고민중에 있습니다! 같이 첨부해주신 이펙티비 자바 아이템도 참고해서 고려해보겠습니다!

해당 부분에서 Pattern에 관련된 부분을 static final로 수정해두었습니다!